hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/BlockReaderFactory.java

```diff
@@ -471,6 +471,10 @@ public ShortCircuitReplicaInfo createShortCircuitReplicaInfo() {
       if (curPeer == null) break;
       if (curPeer.fromCache) remainingCacheTries--;
       DomainPeer peer = (DomainPeer)curPeer.peer;
+      try {
+        peer.setReadTimeout(conf.socketTimeout);
+      }
+      catch(IOException e){}
```

Comment:
According to the javadoc, it should return a null if it is unable to establish a connection. I think it should also return a null in this case as well.